src/extension/api/configuration.ts

```diff
@@ -0,0 +1,64 @@
+import { BehaviorSubject } from 'rxjs'
+import { filter } from 'rxjs/operators'
+import * as sourcegraph from 'sourcegraph'
+import { ClientConfigurationAPI } from '../../client/api/configuration'
+import { ConfigurationCascade } from '../../protocol'
+
+/**
+ * @internal
+ * @template C - The configuration schema.
+ */
+class ExtConfigurationSection<C> implements sourcegraph.Configuration<C> {
+    constructor(private proxy: ClientConfigurationAPI, private data: C) {}
+
+    public get<K extends keyof C>(key: K): C[K] | undefined {
+        return this.data[key]
+    }
+
+    public async update<K extends keyof C>(key: K, value: C[K] | undefined): Promise<void> {
+        await this.proxy.$acceptConfigurationUpdate({ path: [key as string | number], value })
```

Comment:
Why is the cast needed?

Reply:
Because a `keyof C` type can technically be a `symbol` too. We could fix this by using `Extract<keyof C,string|number>` in sourcegraph.d.ts, but that adds a lot of unhelpful complexity to the type signature. Technically it is correct, but the impact is that it would just confuse people because 2% of people know what that means, so it's not appropriate for mass-market public API docs.

I will add a comment explaining this.